scripts/cfa/build_parent_sample.py

```diff
@@ -0,0 +1,315 @@
+    elif args.dataset == "cfa4":
+        info_file = "CFA4_COORDS.txt"
+        file_name = "cfa4.lc.stdsystem.fi.ascii"
+        columns = [
+            "name",
+            "FLT",
+            "time",
+            "N",
+            "sigma_pipe",
+            "sigma_phot",
+            "mag",
+            "mag_err",
+        ]
+    info = {}
+    with open(info_file, "r") as f:
+        for line in f.readlines():
+            info[line.split()[0]] = line.split()[1:]
+    df = pd.read_csv(
+        os.path.join(file_dir, file_name),
+        comment="#",
+        sep=r"\s+",
+        names=columns,
+    )
+    unique_names = set(df["name"])
+    if args.tiny:
+        unique_names = list(unique_names)[:10]
+    for sn_name in unique_names:
+        mask = np.where(df["name"] == sn_name)
+        for key in keys_data:
+            data[key].append(np.array(df[key])[mask])
+        metadata["object_id"].append("SN" + sn_name)
+        if not sn_name.startswith("snf"):
+            metadata["ra"].append(float(info["SN" + sn_name][0]))
+            metadata["dec"].append(float(info["SN" + sn_name][1]))
+        else:
+            metadata["ra"].append(0)
+            metadata["dec"].append(0)
+        metadata["obj_type"].append(" ".join(info["SN" + sn_name][2:]))
+        metadata["redshift"].append(0)
+
+    num_examples = len(metadata["object_id"])
+
+    return num_examples, data, metadata
+
+
+def csp_dr3_bpf(file_dir, data, metadata, keys_data, keys_metadata, args):
+    files = os.listdir(file_dir)
+    num_examples = len(files) - 2  # tab1.dat and SN_photo.dat
+    if args.tiny:
+        num_examples = 10
+        files = files[:10]
+
+    for file in files:
+        if file in ("SN_photo.dat", "tab1.dat"):
+            continue
+        current_filter = None
+        data_ = dict(zip(keys_data, ([] for _ in keys_data)))
+        f = open(os.path.join(file_dir, file), "r")
+        for i, line in enumerate(f.readlines()):
+            if i == 0:
+                # Assuming all are SN Ia. There may be unlabeled subtypes.
+                metadata["obj_type"].append("SN Ia")
+                for key, val in zip(keys_metadata[:-1], line.split()):
+                    if key in ("redshift", "ra", "dec"):
+                        val = float(val)
+                    metadata[key].append(val)
+                continue
+            if line.startswith("filter"):
+                current_filter = line.split()[1]
+                continue
+            for key, val in zip(keys_data[:-1], line.split()):
+                data_[key].append(float(val))
+            data_["FLT"].append(current_filter)
+        for key in keys_data:
+            data[key].append(np.array(data_[key]))
+        f.close()
+    return num_examples, data, metadata
+
+
+survey_specific_logic = {
+    "cfa3": cfa3_bpf,
+    "cfa3_4sh": cfa_generic_bpf,
+    "cfa4": cfa_generic_bpf,
+    "cfa_snII": cfa_snII_bpf,
+}
+
+
+def main(args):
```

Comment:
Listing all arguments instead of grouping them into a single `args` would help in reviewing the logic of the function and detecting any unused or unprovided argument.